### PR TITLE
Issue #24810: properly handle malformed CSV file when setting label

### PR DESCRIPTION
**Pull request by davidbeauchamp**

### CODE REVIEW

File: plugin/csvdata.cpp [hunk 1]
```diff
@@ -226,10 +226,22 @@ void CSVData::setFirstRowHeaders(bool y)
...
 QString CSVData::header(int column)
 {
-  if (_firstRowHeaders && _data && _data->_header.size() >= column)
-    return _data->_header.at(column);
+  QString label;
 
-  return QString::null;
+  if (_firstRowHeaders && _data && _data->_header.size() > column) {
+    label = _data->_header.at(column);
+      if (label.isEmpty()) {
```
Comment (bendiy) [inline plugin/csvdata.cpp]: Indention is off.

File: plugin/csvdata.cpp [hunk 2]
```diff
@@ -226,10 +226,22 @@ void CSVData::setFirstRowHeaders(bool y)
...
+        label = tr("unnamed");
```
Comment (bendiy) [inline plugin/csvdata.cpp]: What about making this:
```
label = tr("unnamed") + QString::number(column);
```
and the same below.

Comment (davidbeauchamp) [inline plugin/csvdata.cpp]: It does that already in the functions that call ::header()

Without first row header:
Without first row header

With first row header:
With first row header

Comment (bendiy) [inline plugin/csvdata.cpp]: Great!

**Comment by bendiy** [comment]
+1

**Pull request merged**